src/DebugEngineHost/HostConfigurationStore.cs

```diff
@@ -88,14 +88,13 @@ public void GetExceptionCategorySettings(Guid categoryId, out HostConfigurationS
         /// </summary>
         /// <param name="enableLoggingSettingName">[Optional] In VS, the name of the settings key to check if logging is enabled. If not specified, this will check 'Logging' in the AD7 Metrics.</param>
         /// <param name="logFileName">[Required] name of the log file to open if logging is enabled.</param>
-        /// <returns>[Optional] If logging is enabled, the logging object.</returns>
+        /// <returns>If no error then logging object. If file cannot be openened then throw an exception. Otherwise return an empty logger - the user can explictly reconfigure it later</returns>
         public HostLogger GetLogger(string enableLoggingSettingName, string logFileName)
         {
             if (string.IsNullOrEmpty(logFileName))
             {
-                throw new ArgumentNullException("logFileName");
+                return new HostLogger();    // empty logger
```

Comment:
This (and below) were to support a scenario where the old-style config caused a logger to already exist and sometime later it was being replaced by command. Only one HostLogger would ever exist, and would (almost) always exist but might not do anything. However I could change it so that the Logger closes the old one and switches to a new HostLogger object when a change is made.